bentoml/_internal/frameworks/tensorflow_v2.py

```diff
@@ -365,19 +347,26 @@ def __call__(self, inputs):
 
         # then save the given model to BentoML modelstore:
         model = NativeModel()
-        tag = bentoml.tensorflow.save("native_toy", model)
+        tag = bentoml.tensorflow.save_model("native_toy", model)
 
     .. note::
 
-       :code:`bentoml.tensorflow.save` API also support saving `RaggedTensor <https://www.tensorflow.org/guide/ragged_tensor>`_ model and Keras model. If you choose to save a Keras model
-       with :code:`bentoml.tensorflow.save`, then the model will be saved under a :obj:`SavedModel` format instead of :obj:`.h5`.
+       :code:`bentoml.tensorflow.save_model` API also support saving `RaggedTensor <https://www.tensorflow.org/guide/ragged_tensor>`_ model and Keras model. If you choose to save a Keras model
+       with :code:`bentoml.tensorflow.save_model`, then the model will be saved under a :obj:`SavedModel` format instead of :obj:`.h5`.
 
     """  # noqa
     context: t.Dict[str, t.Any] = {
         "framework_name": "tensorflow",
         "pip_dependencies": [f"tensorflow=={get_tf_version()}"],
         "import_from_tfhub": False,
     }
+    if signatures is None:
```

Comment:
For each framework:
a. decide if `signatures` is required
b. if there's a common default value for a framework, apply it here as default, but log a warning to tell user about the decision
c. if there's a way to automatically infer the signatures, e.g. with `tf.function`, we will do that if signatures is not provided. and also warn user about the final signatures used.
d. default signature is almost always batchable=False